locksmith/src/utils/middlewares/lockManagerOrPayer.ts

```diff
@@ -0,0 +1,59 @@
+import { RequestHandler } from 'express'
+import networks from '@unlock-protocol/networks'
+import { Web3Service } from '@unlock-protocol/unlock-js'
+import Normalizer from '../normalizer'
+import { SubgraphService } from '@unlock-protocol/unlock-js'
+
+export const lockManagerOrPayerMiddleware: RequestHandler = async (
+  req,
+  res,
+  next
+) => {
+  const lockAddress = Normalizer.ethereumAddress(req.params.lockAddress)
+  const network = Number(req.params.network)
+  const hash = Number(req.params.hash)
+  const userAddress = Normalizer.ethereumAddress(req.user!.walletAddress!)
+
+  if (!lockAddress) {
+    return res.status(404).send({
+      message: 'Missing lock Address',
+    })
+  }
+
+  if (!network) {
+    return res.status(404).send({
+      message: 'Missing network',
+    })
+  }
+
+  const web3Service = new Web3Service(networks)
+
+  const isLockManager = await web3Service.isLockManager(
+    lockAddress,
+    userAddress,
+    network
+  )
+
+  const subgraph = new SubgraphService(networks)
+
+  // get receipt
+  const receipt = await subgraph.receipt(
+    {
+      where: {
+        id: hash,
+        lockAddress,
+        payer: userAddress,
```

Comment:
Let's actually get the receipt by hash only and check after that the payer matches, this will make this code more clear.

Reply:
oh nice! let me clear it